chapters/operatorsandexpressions.tex

```diff
@@ -466,8 +467,19 @@ \subsection{Numeric Functions and Conversion Functions}\label{numeric-functions-
 sqrt($v$)
 \end{lstlisting}\end{synopsis}
 \begin{semantics}
-Square root of $v$ if $v \geq 0$, otherwise an error occurs.
-Argument $v$ needs to be an \lstinline!Integer! or \lstinline!Real! expression.
+Square root of $v$, equivalent to \lstinline!nthRoot(v, 2)!.
+\end{semantics}
+\end{functiondefinition}
+
+\begin{functiondefinition}[nthRoot]
+\begin{synopsis}\begin{lstlisting}
+nthRoot($v$, $n$)
+\end{lstlisting}\end{synopsis}
+\begin{semantics}
+$n$th root of $v$, where $v$ shall be a \lstinline!Real! expression, and $n > 0$ shall be an evaluable \lstinline!Integer! expression.
```

Comment:
```suggestion
$n$th root of $v$, where $v$ shall be a \lstinline!Real! expression, and $n > 0$ shall be an \lstinline!Integer! expression.
```
I don't know if evaluable was a mistake or intentional.

Assuming it was not a mistake - the restrictions of evaluable actually neither accomplishes the intent, nor do I think that it is a good idea.

The main reason I don't think it is a good idea is that we generally don't restrict input variability in this way for normal functions (some operators with function syntax have such restrictions).

That also means that it doesn't really work - inside a function the inputs behave as if they are evaluable expressions, so I can write:
```
function myNthRoot
  input Real x;
  input Integer n;
  output Real y = nthRoot(x, n);
end myNthRoot;
```
And then call `myNthRoot(x, integer(time+1));` completely circumventing the restriction. 

Obviously:
-  `n` must be known to actually evaluate the call - but that is the case for all actual functions
-  non-evaluable integers will cause problems for unit-checking, but that is also the case for `x^n`

Reply:
That limits usability for no obvious user-benefit.

Reply:
The clear user benefit is that we give them unit checking and nice derived units.

Reply:
After giving it some more thought, I now agree that the variability constraint should be removed.  The way I see it working is that it is treated precisely as exponentiation with `Integer` exponent:
- When `v` has empty unit, the unit of `nthRoot(v, n)` is empty (and no variability requirement on `n`).
- Otherwise, `n` must effectively be constant after the stage of parameter evaluation, and the unit of `nthRoot(v, n)` is `v.unit^(1/n)`.
- As long as unit checking isn't generally enforced, don't allow unit checking to trigger evaluation of `n`.

Anyway, this is just the background motivation for proceeding with this PR without a variability constraint on `n`.